### PR TITLE
Issue/11759 ri post card ui improvements

### LINKED ISSUE #11759: Improve UI of post card in the Reader
Update the UI of the post card in the Reader. 

Note: 
- The following design might be out-of-date. Check out `figma` to see the most current design.
Screenshot 2020-04-27 at 12 06 39
Screenshot 2020-04-27 at 12 08 12

### PR DESCRIPTION
**Pull request by ashiagr**
Partially fixes #11759

This PR converts `reader_post_card.xml` to a flat layout hierarchy using constraint layout and introduces UI improvements as per #11759 and latest `Figma` designs for phase 1.

**Light Mode**
| Before | After |
|---|---|
|lightmode_befor| lightmode_after|

**Dark Mode**
| Before | After |
|---|---|
| darkmode_befor| darkmode_after  |

More screenshots

| With disabled icons | Without title | Without image|
|---|---|---|
|disabled_icons| without_title|witout_image|

Notes
- Some of the hidden elements like the attribution layout, thumbnail strip etc. are retained but the design is not updated as it is outside the scope of this PR.
-  Referred Figma "Post Card UI" and "Post Card Actions v1.1" which includes all action icons in the footer and disables instead of hiding them. 
- TBD
    - [ ] New colors under Reader section in `colors.xml`, might want to use existing Semantic colors. 
    - [ ] Spacing between action icons in the footer (currently used constraint layout `spread` [constrain chain](https://developer.android.com/training/constraint-layout/index.html#constrain-chain), advantage - icons will spread appropriately on all devices including tablets. ). Might want to use "spread inside"  with margin for mobile and "spread" for tablet.
    - [ ] Comments action grid-icon without fill color
    - [X] More icon end padding, might want to introduce -ve margin to align with featured image. It is currently displayed in tap-size.   - 982b6a4
    - [X] Moving Visit link to more options - 0a6af83 (as shown in Figma clickable prototype)

PR submission checklist:

- [X] I have considered adding unit tests where possible.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/res/values/colors.xml [hunk 1]
```diff
@@ -309,4 +315,12 @@
     <color name="reader_following_empty_view_outline">@color/gray_10</color>
     <color name="reader_following_empty_view_middle_plus">@android:color/white</color>
     <color name="reader_following_empty_view_background">@android:color/white</color>
+
+    <!--Reader Post Card -->
+    <color name="reader_post_list_background">@color/neutral_0</color>
+    <color name="reader_card_background">@color/white</color>
+    <color name="reader_label_text">#6c7781</color>
```
Comment (zwarm) [inline WordPress/src/main/res/values/colors.xml:322]: @ashiagr Are the reader_label_text and reader_image_featured_border are not in the color palette (as you made reference in the notes)?

Comment (ashiagr) [reply WordPress/src/main/res/values/colors.xml:322]: Yes, I couldn't find in the color palette. I've to discuss it with Shakti.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java [hunk 2]
```diff
@@ -1302,4 +1308,8 @@ protected void onPostExecute(Boolean result) {
             mIsTaskRunning = false;
         }
     }
+
+    private String getUrlWithoutProtocol(String url) {
+        return url.replaceFirst("^(http[s]?://www\\.|http[s]?://|www\\.)", "");
```
Comment (zwarm) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java:1313]: @ashiagr - I might consider writing a unit test for this one if you haven't already.

Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java:1313]: I think we could reuse UrlUtils.removeScheme(...) for this. Wdyt?

Comment (ashiagr) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java]: Reused here: 68c5274

**Comment by ashiagr** [comment]
Thanks for reviewing it @zwarm 🙌. I fixed two small issues from the pending items in this PR. 
982b6a4 & 0a6af83. Can you please review these as well? I'll create another PR for other items to limit this PR size.

**Pull request merged**